docs/vault-k8s: updates for v1.6.0 release (#29152)
hashicorp/vault: website/content/docs/platform/k8s/injector/index.mdx

An example of mounting a Vault Agent configmap [can be found here](/vault/docs/platform/k8s/injector/examples#configmap-example).

### Injector telemetry

> 

Reviewer: ```suggestion

<Tip>

Set [`injector.metrics.enabled`](/vault/docs/platform/k8s/helm/configuration#metrics) 
to `true` in the Helm chart to start collecting injector metrics.

</Tip>
```

> Vault Agent Injector collects the following Prometheus metrics (along with the
> default set of golang metrics):

Reviewer: ```suggestion
Vault Agent injector collects the following Prometheus metrics in addition to
the default set of `golang` metrics:
```

Style correction: do not capitalize features, avoid parentheticals

> - `vault_agent_injector_request_queue_length` - A gauge of webhook requests in
>   the injector's queue.

Reviewer: ```suggestion
- `vault_agent_injector_request_queue_length` - The number of pending webhook requests for the injector.
```

"A gauge" doesn't really describe the metric. Is it the total number of pending requests? The average number of queued requests over an interval?

Author: Sure, "number of pending webhook requests" is clearer.

- `vault_agent_injector_request_processing_duration_ms` - A histogram of webhook
  request processing times in milliseconds.

> - `vault_agent_injector_injections_by_namespace_total` - The total count of
>   Agent container injections by Kubernetes `namespace` and `injection_type`,
>   where `injection_type` is `init_only`, `sidecar_only`, or `init_and_sidecar`.

Reviewer: ```suggestion
- `vault_agent_injector_injections_by_namespace_total` - The total count of
  Agent container injections, grouped by Kubernetes `namespace` and `injection_type`.
  Vault Agent counts the following injection types:
    - `init_only`
    - `sidecar_only`
    - `init_and_sidecar`.
```

Style correction: one idea per sentence

- `vault_agent_injector_failed_injections_by_namespace_total` - The total count

>   of failed Agent Sidecar injections by Kubernetes `namespace`.

Reviewer: ```suggestion
  of failed Agent Sidecar injections, grouped by Kubernetes `namespace`.
```

> Enable metrics collection in the injector by setting [`injector.metrics.enabled:
> true`](/vault/docs/platform/k8s/helm/configuration#metrics) in the Helm chart.

Reviewer: ```suggestion
```
Suggest moving this to the start of the section so folks can find it easier/faster.
